### PR TITLE
bpo-39868: Update Language Reference for PEP 572.

### PR DESCRIPTION
**Pull request by brandtbucher**
Leaving the prose part blank for @shankarj67 to fill in a follow-up PR.

https://bugs.python.org/issue39868

### CODE REVIEW

**Comment by brandtbucher** [comment]
I think this should be good, but perhaps you can double-check my grammar here @gvanrossum? You likely have a better understanding of PEP 572's corner-cases than I do.

File: Doc/reference/compound_stmts.rst [hunk 1]
```diff
@@ -90,8 +90,8 @@ The :keyword:`!if` statement
 The :keyword:`if` statement is used for conditional execution:
 
 .. productionlist::
-   if_stmt: "if" `expression` ":" `suite`
-          : ("elif" `expression` ":" `suite`)*
+   if_stmt: "if" `assignment_expression` ":" `suite`
+          : ("elif" `assignment_expression` ":" `suite`)*
           : ["else" ":" `suite`]
 
 It selects exactly one of the suites by evaluating the expressions one by one
```
Comment (gvanrossum) [inline Doc/reference/compound_stmts.rst:97]: Do you think "expression" here needs updating too? It becomes a bit ugly that way, and in context it doesn't seem ambiguous.

**Review by gvanrossum — APPROVED** [review]
Looks good. I'll merge this now. For the TODO part we'll wait for Shankar Jha (see bpo-39868). Please give him some guidance.

File: Doc/reference/compound_stmts.rst (hunk 1 above)
Comment (gvanrossum) [reply Doc/reference/compound_stmts.rst:97]: (I thought more about it and I don't think it needs updating, so I'll land now.)

**Comment by brandtbucher** [comment]
Thanks, sounds good.

**Pull request merged**